Makefile

```diff
@@ -39,6 +39,21 @@ else
 	RUNTIME_DEPS_TESTEXEC=true
 endif
 
+# MACOS_SIGN_AND_NOTARIZE: When false, creates a release tarball as part of the make process,
```

Comment:
One thing that's confusing to me about these changes is that even though the Makefile has a MACOS_SIGN_AND_NOTARIZE variable, if I set it to true, my binaries are not going to be notarized :) All of that happens elsewhere, in GHA pipelines, right?

I almost think it might be more clear to not change this file... We can leave it as-is, right? It's not like producing the tarball deletes the unsigned binaries from the cache...

Reply:
OK actually I think this variable might never be set to true, since it looks like `sign-and-notarize.sh` unpacks and repacks the tarball. So it seems like we could definitely leave as-is.